File: Back-End/models/agent.js
const { nanoid } = require('nanoid');

module.exports = (sequelize, DataTypes) => {
    return sequelize.define('Agent', {
        agentID: {
            type: DataTypes.STRING,
            primaryKey: true,
            defaultValue: () => `agt_${nanoid()}`,
        },
        name: { type: DataTypes.STRING },
        lastname: { type: DataTypes.STRING},
        wallet: { type: DataTypes.STRING },
        cin: { type: DataTypes.STRING, unique: true},
        email: { type: DataTypes.STRING, unique: true },
        phone: { type: DataTypes.STRING },
        location: { type: DataTypes.STRING},
    });
};

File: Mobile/lib/models/agent.dart
// models/agent.dart
class Agent {
  final String? agentID;
  final String? name;
  final String? lastname;
  final String? cin;
  final String? email;
  final String? phone;
  final String? wallet;
  final String? location;

  Agent({
    this.agentID,
    this.name,
    this.lastname,
    this.cin,
    this.email,
    this.phone,
    this.wallet,
    this.location,
  });

  // Convert JSON to Agent object
  factory Agent.fromJson(Map<String, dynamic> json) {
    return Agent(
      agentID: json['agentID'],
      name: json['name'],
      lastname: json['lastname'],
      cin: json['cin'],
      email: json['email'],
      phone: json['phone'],
      wallet: json['wallet'],
      location: json['location'],
    );
  }

  // Convert Agent object to JSON
  Map<String, dynamic> toJson() {
    return {
      'agentID': agentID,
      'name': name,
      'lastname': lastname,
      'cin': cin,
      'email': email,
      'phone': phone,
      'wallet': wallet,
      'location': location,
    };
  }
}